src/components/svg/wrap/svg-keyboard-wrap.component.ts

```diff
@@ -0,0 +1,156 @@
+            state('leftIn', style({
+                transform: 'translateX(-50%)',
+                left: '50%'
+            })),
+            state('leftOut', style({
+                transform: 'translateX(-100%)',
+                left: '0'
+            })),
+            /* Right -> Left animation */
+            state('rightIn', style({
+                transform: 'translateX(-50%)',
+                left: '50%'
+            })),
+            state('rightOut', style({
+                transform: 'translateX(0%)',
+                left: '100%'
+            })),
+            /* Transitions */
+            transition('none => leftIn, leftOut => leftIn', [
+                style({
+                    opacity: 0,
+                    transform: 'translateX(0%)',
+                    left: '100%'
+                }),
+                style({
+                    opacity: 1
+                }),
+                animate('400ms ease-out')
+            ]),
+            transition('* => none', [
+                style({
+                    opacity: 0,
+                    transform: 'translateX(-100%)',
+                    left: '0'
+                }),
+                style({
+                    opacity: 1
+                })
+            ]),
+            transition('none => rightIn, rightOut => rightIn', [
+                style({
+                    opacity: 0,
+                    transform: 'translateX(-100%)',
+                    left: '0'
+                }),
+                style({
+                    opacity: 1
+                }),
+                animate('400ms ease-out')
+            ]),
+            transition(
+                'leftIn => leftOut,' +
+                'rightIn => rightOut,' +
+                'leftIn <=> rightOut,' +
+                'rightIn <=> leftOut',
+                animate('400ms ease-out')
+            )
+        ])
+    ]
+})
+export class SvgKeyboardWrapComponent implements OnInit {
+    @Input() layers: Layer[];
+    @Input() popoverEnabled: boolean = true;
+    @Input() animationEnabled: boolean = true;
+
+    private popoverShown: boolean;
+    private keyEditConfig: { moduleId: number, keyId: number };
+    private popoverInitKeyAction: KeyAction;
+    private module: Module[];
+    private currentLayer: number;
+
+    constructor() {
+        this.keyEditConfig = {
+            moduleId: undefined,
+            keyId: undefined
+        };
+
+        this.currentLayer = 0;
+    }
+
+    ngOnInit() {
+        this.layers[0].animation = 'leftIn';
+    }
+
+    ngOnChanges() {
+        this.currentLayer = 0;
+        this.layers.map((layer: Layer) => layer.animation = 'none');
+        this.layers[0].animation = 'leftIn';
+    }
+
+    onKeyClick(moduleId: number, keyId: number, module: Module[]): void {
+        if (!this.popoverShown) {
+            this.keyEditConfig = {
+                moduleId,
+                keyId
+            };
+
+            this.module = module;
```

Comment:
How about changing this line to `let modules: Module[] = this.layers[this.currentLayer].modules;` ?